### PR TITLE
[colotensor] add Tensor.view op and its unit test

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: colossalai/nn/_ops/view.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        raise RuntimeError("shape '{}' is invalid for input of size {}".format(tgt_sp, org_prod))
+
+    infer_dim = -(org_prod // tgt_prod)
+    return tgt_sp[: pos] + (infer_dim,) + tgt_sp[pos + 1:]
+
+
+@colo_op_impl(torch.Tensor.view)
+def colo_view(self: ColoTensor, *shape) -> 'ColoTensor':
+    """Handles ``__torch_function__`` dispatch for ``torch.Tensor.view``.
+    Changes the shape current tensor.
```
Comment (feifeibear) [inline colossalai/nn/_ops/view.py:57]: Changes the shape of the current tensor.

File: colossalai/nn/_ops/view.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    """
+    assert isinstance(self, ColoTensor)
+    # apply original `view` function for replicated colo tensors
```
Comment (feifeibear) [inline colossalai/nn/_ops/view.py:60]: if torch tensor
return self.view()?

File: tests/test_tensor/ops/test_view.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    split_param_row_tp1d(x, pg)
+
+    x.view('a', 'b', 'c')
+    x.view(8, -1)
+    x.view([-2, -2, -2])
+    x.view((-1, -1, -1))
+
+
+def run_dist(rank, world_size, port):
+    config = dict(parallel=dict(tensor=dict(mode="1d", size=world_size),))
```
Comment (feifeibear) [inline tests/test_tensor/ops/test_view.py:84]: dict = {}
I think you do not need to use gpc

File: tests/test_tensor/ops/test_view.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+
+def exam_view_errors(pg):
+    x = torch.randn(8, 2, device=get_current_device())
+    x = ColoTensor(x, ColoTensorSpec(pg))
+    split_param_row_tp1d(x, pg)
+
+    x.view('a', 'b', 'c')
+    x.view(8, -1)
+    x.view([-2, -2, -2])
```
Comment (feifeibear) [inline tests/test_tensor/ops/test_view.py:79]: What is the purpose of these code?
Using view in this way is invalid?
```
>>> import torch
>>> a = torch.randn(8, 2)
>>> a
tensor([[-0.1541,  0.6496],
        [-0.1736,  1.0077],
        [ 0.1969,  0.2545],
        [-0.3447, -3.6378],
        [-0.8737, -0.3517],
        [-0.6656,  0.7300],
        [ 0.1359, -1.5623],
        [-0.0823,  0.0673]])
>>> a.view([-2, -2, -2])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: invalid shape dimension -2
```

File: colossalai/tensor/compute_spec.py [hunk 5]
```diff
@@ -22,4 +22,7 @@ def __init__(self, compute_pattern: ComputePattern) -> None:
         self.output_replicate = True
 
     def __repr__(self):
-        return f'compute pattern: {self.compute_pattern}'
+        return f'Compute pattern: {self.compute_pattern}'
+
+    def set_keeping_output(self):
```
Comment (feifeibear) [inline colossalai/tensor/compute_spec.py:27]: Not used!
signature had better to be set_output_replicate(self, flag : bool)

File: colossalai/nn/_ops/view.py (hunk 2 above)
Comment (1SAA) [reply colossalai/nn/_ops/view.py:60]: If `self` is a instance of torch.Tensor, this function will not be triggered.

File: colossalai/tensor/compute_spec.py (hunk 5 above)
Comment (1SAA) [reply colossalai/tensor/compute_spec.py:27]: It will be used in an upcoming unit test right away.

File: colossalai/nn/_ops/view.py (hunk 1 above)
Comment (1SAA) [reply colossalai/nn/_ops/view.py:58]: Done

File: tests/test_tensor/ops/test_view.py (hunk 3 above)
Comment (1SAA) [reply tests/test_tensor/ops/test_view.py:84]: Done

**Review by feifeibear — APPROVED** [review]

**Pull request merged**